src/SafeArrayIterator.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace MaxGoryunov\SavingIterator\Src;
+
+use ArrayAccess;
+use Countable;
+use Iterator;
+use Serializable;
+
+/**
+ * Safe array iterator. Copies array when cloned.
+ * @template TKey
+ * @template TValue
+ * @implements ArrayAccess<TKey, TValue>
+ * @implements Iterator<TKey, TValue>
+ * 
+ * @since 0.3
+ */
+final class SafeArrayIterator implements ArrayAccess, Countable, Iterator, Serializable
```

Comment:
`SafeArrayIterator` implements all interface methods of `ArrayIterator` and acts as a safe alternative to it. It is called Safe because when it is cloned its array of stored values is copied as well.